internal/types/types.go

```diff
@@ -1,4 +1,4 @@
-package specutil
+package types
```

Comment:
Package `types` is too vague, and I'm not sure it belongs to this namescape. It should reside in the HCL namespace in my opinion, because other languages (e.g. TypeScript) already have a type system. I'd also love to see how we can merge this logic into one of the existing packages because the logic of `schemaspec` spreads across too many atm.